### PR TITLE
document the GC stack size user option

### PR DESCRIPTION
**Pull request by frivieremicroej**

### CODE REVIEW

File: ApplicationDeveloperGuide/sectionArchitectureOptions.rst [hunk 1]
```diff
@@ -821,6 +821,37 @@ Enables :ref:`soar_binary_code_verifier` during application build.
...
+You can receive a notification when the stack limit is reached by implementing the following hook:
+
+.. code-block:: c
+
+   void LLMJVM_on_GC_MarkStackOverflow_reached(void) {
+      // When entering here, the GC stack is undersized, which may affect GC performance.
+      // It is recommended to either increase the GC stack size or reduce the object graph depth.
+   }
+
+using . When the maximum of stack is reached,  depth-first search of alive object GC stack size
```
Comment (jerome-leroux) [inline ApplicationDeveloperGuide/sectionArchitectureOptions.rst:854]: ?
```suggestion

```

File: ApplicationDeveloperGuide/sectionArchitectureOptions.rst [hunk 2]
```diff
@@ -821,6 +821,37 @@ Enables :ref:`soar_binary_code_verifier` during application build.
...
+Option(text): Stack Size
+""""""""""""""""""""""""
+
+*Option Name*: ``com.microej.runtime.core.gc.stacklevels.max``
+
+*Default value*: ``32``
+
+*Description*:
+
+Indicates the quantity of items in the :ref:`Garbage Collector <runtime_gc>`'s internal stack.
```
Comment (jerome-leroux) [inline ApplicationDeveloperGuide/sectionArchitectureOptions.rst:840]: Some idea to name the stack:
- frontier stack? see https://medium.com/tebs-lab/breadth-first-search-and-depth-first-search-4310f3bf8416
- LIFO buffer?

**Review by jerome-leroux — CHANGES_REQUESTED** [review]

File: ApplicationDeveloperGuide/sectionArchitectureOptions.rst (hunk 1 above)
Comment (frivieremicroej) [reply ApplicationDeveloperGuide/sectionArchitectureOptions.rst]: removed

File: ApplicationDeveloperGuide/sectionArchitectureOptions.rst (hunk 2 above)
Comment (frivieremicroej) [reply ApplicationDeveloperGuide/sectionArchitectureOptions.rst:840]: I understand the potential conflict between threads stack and GC stack (GC is not a thread), but I don't think these proposals are more appropriate.
What about "GC working buffer" ?

Comment (jerome-leroux) [reply ApplicationDeveloperGuide/sectionArchitectureOptions.rst:840]: Why not just calling it "GC Mark Stack"?

See the following documents that talk about the same issue:
- https://www.ibm.com/docs/en/ztpf/1.1.0.15?topic=collection-mark-phase
- https://www.ibm.com/docs/en/ztpf/1.1.0.15?topic=phase-mark-stack-overflow
- https://www.ibm.com/support/pages/preventing-mark-stack-overflows-during-garbage-collection-java-sdk

**Review by jerome-leroux — APPROVED** [review]

**Pull request merged**